lib/srv/server/cloudwatcher.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/cloudflare/cfssl/log"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/cloud"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+	"github.com/gravitational/trace"
+)
+
+type EC2Instances struct {
+	Region    string
+	Document  string
+	Instances []*ec2.Instance
+}
+
+type Watcher struct {
+	// Instances can be used to consume
+	Instances chan EC2Instances
+
+	fetchers []*ec2InstanceFetcher
+	waitTime time.Duration
+	ctx      context.Context
+	cancel   context.CancelFunc
+}
+
+func (w *Watcher) Run() {
+	ticker := time.NewTicker(w.waitTime)
+	for {
+		for _, fetcher := range w.fetchers {
+			inst, err := fetcher.GetEC2Instances(w.ctx)
+			if err != nil {
+				log.Error("Failed to fetch EC2 instances: ", err)
```

Comment:
When the agent has no permission to fetch, this log will spam every time it tries. In the db cloud fetcher, we have downgraded `AccessDenied` errors to a debug log.